coding/reader.hpp

```diff
@@ -19,10 +20,12 @@ class Reader
   DECLARE_EXCEPTION(SizeException, Exception);
   DECLARE_EXCEPTION(ReadException, Exception);
 
+  using TReaderPtr = unique_ptr<Reader>;
```

Comment:
Generally I am against type aliases for smart pointers as they don't give any clue about what smart pointer if any is used here. Please, use unique ptr in function signatures and auto for variables instead.